relay/blockchain/exchange_proxy.py

```diff
@@ -49,14 +62,56 @@ def log(log_entry):
             f(log_entry['args']['orderHash'],
               log_entry['args']['filledMakerTokenAmount'],
               log_entry['args']['filledTakerTokenAmount'])
-        self.start_listen_on(FillEvent, log)
+        self.start_listen_on(LogFillEventType, log)
 
     def start_listen_on_cancel(self, f) -> None:
         def log(log_entry):
             f(log_entry['args']['orderHash'],
               log_entry['args']['cancelledMakerTokenAmount'],
               log_entry['args']['cancelledTakerTokenAmount'])
-        self.start_listen_on(CancelEvent, log)
+        self.start_listen_on(LogCancelEventType, log)
+
+    def get_exchange_events(self,
```

Comment:
I noticed that we have a lot of duplication in those functions. But I will address that in a separate refactoring pull, so in this PR it can stay like that.